WordPress/gutenberg
# CustomSelectControl: Hard deprecate constrained width

## What?

Promotes `__nextUnconstrainedWidth` on CustomSelectControl to the default behavior.

## Why?

The deprecation grace period has elapsed.

## Testing Instructions

Unfortunately, this compatibility mode had broken at some point, so you won't even be able to see this width switching behavior on [Storybook trunk](https://wordpress.github.io/gutenberg/?path=/docs/components-customselectcontrol--docs) 😱

If you build Storybook on this PR branch, you should be able to see that:

- CustomSelectControl renders without a constrained width, i.e. fills the entire available width.
- It should no longer log the deprecation warning.
- The `__nextUnconstrainedWidth` prop is removed from the docs.


## ✍️ Dev Note

`CustomSelectControl` used to have a hard-coded width, which was inconsistent with our other form components. In WordPress 6.1, we deprecated this unconstrained width, and introduced the transitional `__nextUnconstrainedWidth` prop so consumers could opt into these new styles before they become the default.

In WordPress 6.5, these unconstrained width styles have become the default. Any use of the `__nextUnconstrainedWidth` prop can be safely removed.
